event/bind.go

```diff
@@ -1,71 +1,122 @@
+
+// UnsafeBind registers a callback function to be called whenever the provided event is triggered
+// against this bus. The binding is concurrently bound, and therefore may not be immediately
+// available to be triggered. When Reset is called on a Bus, all prior bindings are unbound. This
+// call is 'unsafe' because UnsafeBindables use bare interface{} types.
+func (bus *Bus) UnsafeBind(eventID UnsafeEventID, callerID CallerID, fn UnsafeBindable) Binding {
+	bindID := BindID(atomic.AddInt64(bus.nextBindID, 1))
+	go func() {
+		bus.mutex.Lock()
+		bus.getBindableList(eventID, callerID).storeBindable(fn, bindID)
+		bus.mutex.Unlock()
+	}()
+	return Binding{
+		Handler:  bus,
+		EventID:  eventID,
+		CallerID: callerID,
+		BindID:   bindID,
+	}
 }
 
-// PersistentBind acts like Bind, but persists the binding such that if the event
-// bus is reset, the binding will still trigger. Thes bindings should likely be global
-// bindings, using a CID of 0, or be tolerant to the CID bound not being present after
-// such a clear.
-func (eb *Bus) PersistentBind(name string, callerID CID, fn Bindable) {
-	eb.pendingMutex.Lock()
-	opt := UnbindOption{
-		Event: Event{
-			Name:     name,
-			CallerID: callerID,
-		}, Fn: fn}
-	eb.binds = append(eb.binds, opt)
-	eb.persistentBinds = append(eb.persistentBinds, opt)
-	eb.pendingMutex.Unlock()
+// PersistentBind acts like UnsafeBind, but cause Bind to be called with these inputs after a Bus is Reset, i.e.
+// persisting the binding through bus resets. Unbinding this will not stop it from being rebound on the next
+// Bus Reset-- ClearPersistentBindings will.
+func (bus *Bus) PersistentBind(eventID UnsafeEventID, callerID CallerID, fn UnsafeBindable) Binding {
+	binding := bus.UnsafeBind(eventID, callerID, fn)
+	go func() {
+		bus.mutex.Lock()
+		bus.persistentBindings = append(bus.persistentBindings, persistentBinding{
+			eventID:  eventID,
+			callerID: callerID,
+			fn:       fn,
+		})
+		bus.mutex.Unlock()
+	}()
+	return binding
 }
 
-// ClearPersistentBindings removes all persistent bindings. It will not unbind them
-// from the bus, but they will not be bound following the next bus reset. 
-func (eb *Bus) ClearPersistentBindings() {
-	eb.pendingMutex.Lock()
-	eb.persistentBinds = []UnbindOption{}
-	eb.pendingMutex.Unlock()
+// Unbind unregisters a binding from a bus concurrently. Once complete, triggers that would
+// have previously caused the Bindable callback to execute will no longer do so.
+func (bus *Bus) Unbind(loc Binding) {
+	go func() {
+		bus.mutex.Lock()
+		bus.getBindableList(loc.EventID, loc.CallerID).remove(loc.BindID)
+		bus.mutex.Unlock()
+	}()
 }
 
-// GlobalBind binds on the bus to the cid 0, a non entity.
-func (eb *Bus) GlobalBind(name string, fn Bindable) {
-	eb.Bind(name, 0, fn)
+// A Bindable is a strongly typed callback function to be executed on Trigger. It must be paired
+// with an event registered via RegisterEvent.
+type Bindable[Payload any, C any] func(C, Payload) Response
```

Comment:
I hate to say it but I think we should change this to not flip args.
When newer implementers are using this the error messages are a bit confusing due to the flip. 
Take for example this error message I got while going through the click propagation example:
.\main.go:56:48: type func(me *"github.com/oakmound/oak/v3/mouse".Event, box *hoverButton) event.Response of func(me *mouse.Event, box *hoverButton) event.Response {…} does not match inferred type event.Bindable[*"github.com/oakmound/oak/v3/mouse".Event, *hoverButton] for event.Bindable[vent, *hoverButton] for event.Bindable[Payload, C]

After writing a few its a pretty well known paradigm but it feels like if we are to make this switch it would be now.

Reply:
To clarify, you're suggesting:
```
--- a/event/bind.go
+++ b/event/bind.go
@@ -81,9 +81,9 @@ func (bus *Bus) Unbind(loc Binding) {

 // A Bindable is a strongly typed callback function to be executed on Trigger. It must be paired
 // with an event registered via RegisterEvent.
-type Bindable[Payload any, C any] func(C, Payload) Response
+type Bindable[C any, Payload any] func(C, Payload) Response

-func Bind[Payload any, C Caller](b Handler, ev EventID[Payload], c C, fn Bindable[Payload, C]) Binding {
+func Bind[C Caller, Payload any](b Handler, ev EventID[Payload], c C, fn Bindable[C, Payload]) Binding {
        return b.UnsafeBind(ev.UnsafeEventID, c.CID(), func(c CallerID, h Handler, payload interface{}) Response {
                typedPayload := payload.(Payload)
                ent := h.GetCallerMap().GetEntity(c)
```

Reply:
Exactly!
How do you feel about that rewrite?

Reply:
It's a good improvement; it won't require any refactoring either, as the type arguments should be implied from the inputs.